pennylane/devices/qubit/initialize_state.py

```diff
@@ -40,8 +40,13 @@ def create_initial_state(
     """
     if not prep_operation:
         num_wires = len(wires)
-        state = np.zeros((2,) * num_wires)
+        state = np.zeros((2,) * num_wires, dtype=complex)
```

Comment:
I think this is fine for now, but we should mark this to respect the precision of the input params at this stage rather than doing a cast later, as it will be much more expensive for large data sets.